eZ/Publish/API/Repository/Tests/UserServiceTest.php

```diff
@@ -2553,4 +2554,53 @@ private function createMultiLanguageUser($userGroupId = 13)
         // Create a new user
         return $userService->createUser($userCreateStruct, [$group]);
     }
+
+    /**
+     * Test for the createUser() method.
+     *
+     * @see \eZ\Publish\API\Repository\UserService::createUser()
+     */
+    public function testCreateUserInvalidPasswordHashTypeThrowsException()
+    {
+        $repository = $this->getRepository();
+        $signalSlotUserService = $repository->getUserService();
+
+        $signalSlotUserServiceReflection = new ReflectionClass($signalSlotUserService);
+        $userServiceProperty = $signalSlotUserServiceReflection->getProperty('service');
+        $userServiceProperty->setAccessible(true);
+        $userService = $userServiceProperty->getValue($signalSlotUserService);
+
+        $userServiceReflection = new ReflectionClass($userService);
+        $settingsProperty = $userServiceReflection->getProperty('settings');
+        $settingsProperty->setAccessible(true);
+
+        $defaultUserServiceSettings = $settingsProperty->getValue($userService);
+
+        /* BEGIN: Use Case */
+        $settingsProperty->setValue(
+            $userService,
+            [
+                'hashType' => 42424242, // Non-existing hash type
+            ] + $settingsProperty->getValue($userService)
+        );
+
+        try {
+            $this->createUserVersion1();
+        } catch (InvalidArgumentException $e) {
+            $this->assertEquals(
+                "Argument 'type' is invalid: Password hash type '42424242' is not recognized",
+                $e->getMessage());
+
+            // Reset to default settings, so we don't break other tests
+            $settingsProperty->setValue($userService, $defaultUserServiceSettings);
+
+            return;
+        }
+        /* END: Use Case */
+
+        // Reset to default settings, so we don't break other tests
+        $settingsProperty->setValue($userService, $defaultUserServiceSettings);
+
+        $this->fail('Did not throw an InvalidArgumentException.');
```

Comment:
New best practices: Please use `$this->expectException(InvalidArgumentException::class)` at the beginning of the test instead of this line.

Reply:
@alongosz It doesn't seem to work. Is it because I catch the exception in the test? I have to do that, because if I don't I can't reset the settings, and other tests will fail further down the line. I can re-throw after resetting of course, but is that still best practice? Feels kind of pointless.

Reply:
Well... that's because test is hack-ish :)
You can add at the top:
`$this->expectExceptionMessage("Argument 'type' is invalid: Password hash type '42424242' is not recognized");`

and catch modify to:
```php
} catch (InvalidArgumentException $e) {
            // Reset to default settings, so we don't break other tests
            $settingsProperty->setValue($userService, $defaultUserServiceSettings);

            throw $e;
}
```

Reply:
Yep, that's what I just did. Seems it has to be either hackish or more hackish. But if it works...

Reply:
yeah, reflection is not really supposed to be in integration tests

Reply:
But ok, you have your approvals, lets send to QA